libs/vkd3d/command.c

```diff
@@ -17606,11 +17608,42 @@ static CONST_VTBL struct ID3D12CommandQueueVtbl d3d12_command_queue_vtbl =
     d3d12_command_queue_GetDesc,
 };
 
+static bool d3d12_command_queue_needs_cpu_waits_locked(struct d3d12_command_queue *command_queue)
+{
+    uint64_t current_time_ns, queue_submit_time_ns;
+    unsigned int i;
+
+    if (command_queue->vkd3d_queue->command_queue_count == 1)
+        return false;
+
+    current_time_ns = vkd3d_get_current_time_ns();
+
+    for (i = 0; i < command_queue->vkd3d_queue->command_queue_count; i++)
+    {
+        struct d3d12_command_queue *q = command_queue->vkd3d_queue->command_queues[i];
+
+        if (q == command_queue)
+            continue;
+
+        /* If any other virtual queue is actively doing submissions, resolve waits
+         * on the CPU in order to avoid delays caused by false dependencies. */
+        queue_submit_time_ns = vkd3d_atomic_uint64_load_explicit(&q->last_submission_time_ns, vkd3d_memory_order_relaxed);
```

Comment:
Need explicit alignment for last_submission_time_ns on 32-bit.

Reply:
made it `UINT64`.